stdlib/src/utils/string_slice.mojo

```diff
@@ -371,11 +375,7 @@ struct StringSlice[is_mutable: Bool, //, origin: Origin[is_mutable].type,](
         Args:
             value: The string value.
         """
-
-        debug_assert(
-            _is_valid_utf8(value.as_bytes()), "value is not valid utf8"
-        )
-        self = StringSlice[O](unsafe_from_utf8=value.as_bytes())
+        self = StringSlice[O](unsafe_from_utf8=value.as_bytes[False, O]())
```

Comment:
**Suggestion** Here and below it may be nicer to say `is_mutable=False` to help clarify what's going on here.  Thoughts?

Reply:
I think it is a bit self-evident with `O: ImmutableOrigin`